receiver/promtailreceiver/config.go

```diff
@@ -52,9 +52,9 @@ func NewConfigWithID(operatorID string) *Config {
 }
 
 type PromtailInputConfig struct {
-	PositionsConfig positions.Config      `mapstructure:"positions,omitempty" yaml:"positions,omitempty"`
-	ScrapeConfig    []scrapeconfig.Config `mapstructure:"scrape_configs,omitempty" yaml:"scrape_configs,omitempty"`
-	TargetConfig    file.Config           `mapstructure:"target_config,omitempty" yaml:"target_config,omitempty"`
+	PositionsConfig positions.Config      `mapstructure:"-" yaml:"positions,omitempty"`
+	ScrapeConfig    []scrapeconfig.Config `mapstructure:"-" yaml:"scrape_configs,omitempty"`
+	TargetConfig    file.Config           `mapstructure:"-" yaml:"target_config,omitempty"`
 }
```

Comment:
Why this change?

Reply:
The test https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/54da8bd95661636d0639b1b2bd2832a391d406ee/cmd/otelcontribcol/components_test.go#L12 failed because inner structs don't have `mapstructure` tags, they only have `yaml` tags.

Since promtail config has its own Unmarshall method that uses `yaml.Marshall`, `yaml.Unmarshall` methods, `mapstructure` tag is not needed here